cmd/thanos/main.go

```diff
@@ -73,7 +75,7 @@ func main() {
 	registerStore(cmds, app, "store")
 	registerQuery(cmds, app, "query")
 	registerRule(cmds, app, "rule")
-	registerCompact(cmds, app, "compact")
+	registerCompact(cmds, app, component.Compact)
```

Comment:
I know that you have not introduced this parameter but I feel that in the long term this parameter should go away and we should just use `component.Compact` inside of the function itself since it is already designated for registering the `compact` command :smile:

Reply:
Sounds good to me, I thought about it too but I wasn't sure even with this change.
If we agree I'll be happy to change it. PR like this will come for every component so eventually I can change it for all of them.

Reply:
changed in https://github.com/improbable-eng/thanos/pull/1297/commits/127ee71d2a18dd59d7f1606ceceb943db969c094 is this ok with you?